pkg/apis/application/v1alpha1/types.go

```diff
@@ -170,6 +173,34 @@ type ApplicationSource struct {
 	Plugin *ApplicationSourcePlugin `json:"plugin,omitempty" protobuf:"bytes,11,opt,name=plugin"`
 	// Chart is a Helm chart name, and must be specified for applications sourced from a Helm repo.
 	Chart string `json:"chart,omitempty" protobuf:"bytes,12,opt,name=chart"`
+	// Ref is reference to another source within sources field. This field will not be used if used with a `source` tag.
+	Ref string `json:"ref,omitempty" protobuf:"bytes,13,opt,name=ref"`
+}
+
+// ApplicationSources contains list of required information about the sources of an application
+type ApplicationSources []ApplicationSource
+
+func (a *ApplicationSpec) GetSource() ApplicationSource {
+	// if Application has multiple sources, return the first source in sources
+	if a.Sources != nil && len(a.Sources) > 0 {
+		return a.Sources[0]
+	}
+	if a.Source != nil {
+		return *a.Source
+	}
+	return ApplicationSource{}
+}
+
+func (a *ApplicationSpec) HasMultipleSources() bool {
+	return a.Sources != nil && len(a.Sources) > 0
+}
+
+func (a *ApplicationSpec) GetSourcePtr() *ApplicationSource {
+	// if Application has multiple sources, return the first source in sources
+	if a.Sources != nil && len(a.Sources) > 0 {
```

Comment:
Noted that you added a HasMultipleSources() method, maybe use that here instead?